dipy/segment/clusteringspeed.pyx

```diff
@@ -582,7 +582,7 @@ cdef class QuickBundles(object):
         nearest_cluster = self.find_nearest_cluster(self.features)
 
         # Find nearest cluster to s_i_flip if metric is not order invariant
-        if not self.metric.feature.is_order_invariant:
+        if self.metric.feature.is_order_invariant != 0:
```

Comment:
I think the test was good before when I look at the docstring just above. Can you explain to us what is wrong?

Reply:
This code in the if-block is not executed in case of a non-order_invariant metric like AveragePointwiseEuclideanMetric.

I have added a test that hopefully proves the point :)

Reply:
@koendc your change is inverting the test condition, i.e. only compute the flip distance if the metric **is** order-invariant. This is not what it should be doing, we want to compute the flip when it **is not** order-invariant. Otherwise, it is just wasting computation as the results between flip and non-flip should be the same.